pkg/rng/entropy_linux.go

```diff
@@ -118,45 +121,53 @@ func UpdateLinuxRandomness(recoverer recovery.Recoverer) error {
 	if err != nil {
 		return err
 	}
+	defer hwRng.Close()
 
 	rng, err := os.OpenFile(RandomDevice, os.O_APPEND|os.O_WRONLY, os.ModeDevice)
 	if err != nil {
 		return err
 	}
+	defer rng.Close()
 
-	go func() {
-		defer hwRng.Close()
-		defer rng.Close()
-
-		for {
-			time.Sleep(EntropyFeedTime)
+	for {
+		time.Sleep(EntropyFeedTime)
```

Comment:
Why did there used to be a goroutine here, and why is it now gone?

Reply:
There was a goroutine here before because the loop was never meant to be terminated and just ran endlessly until the program got terminated externally. This behavior was very hard to test to I added (in my opinion) sane break conditions to the loop. This also makes the goroutine unnecessary as the loop won't run forever now.

Reply:
Yeah, I'm confused about it honestly, but what you did changed the whole purpose of the function.

But in looking up how it's used I discovered something: it's not used anywhere. Why don't we just delete it altogether?